src/main/java/au/org/ala/biocache/dao/SearchDAOImpl.java

```diff
@@ -717,11 +852,12 @@ public void writeCoordinatesToStream(SearchRequestParams searchParams,OutputStre
      * @param includeSensitive
      * @throws Exception
      */
-    public Map<String, Integer> writeResultsFromIndexToStream(final DownloadRequestParams downloadParams,
+    public ConcurrentMap<String, AtomicInteger> writeResultsFromIndexToStream(final DownloadRequestParams downloadParams,
                                                                          OutputStream out,
                                                                          boolean includeSensitive, final DownloadDetailsDTO dd, boolean checkLimit) throws Exception {
+        ExecutorService nextExecutor = getSolrThreadPoolExecutor();
```

Comment:
Use separate pools for online downloads and offline downloads. 

Online downloads have dd == null || dd.getFileLocation() == null